client/src/main/scala/org/http4s/client/middleware/GZip.scala

```diff
@@ -58,15 +59,15 @@ object GZip {
         response.withBodyStream(response.body.through(decompressWith(gunzip)))
 
       case Some(header) if header.contentCoding == ContentCoding.deflate =>
-        val deflate: Pipe[F, Byte, Byte] = fs2.compression.deflate(bufferSize)
+        val deflate: Pipe[F, Byte, Byte] = fs2.compression.deflate(DeflateParams(bufferSize))
```

Comment:
Maybe we should surface the `DeflateParams` at the top level to control more than buffer size?

Reply:
The only trouble here is that the choice of zlib/gzip are also selected via `DeflateParams`, so we'd want it to be `zlib` here specifically. We could overwrite but I don't know if that's too dirty..

Reply:
Oh.  Maybe we deprecate the old one and just have a more general `Compression` middleware?

Reply:
I'll track this in another issue to think about later :)